proposals/2312-matrix-uri.md

```diff
@@ -356,6 +356,95 @@ Client authors are strongly encouraged to standardise custom query elements
 that gain adoption by submitting an MSC defining them in a way compatible
 across the client ecosystem.
 
+We define four other query parameters below which improve the ability to onboard
+new matrix users (`default_client` and `default_hs`) and let link authors
+specify a mandated client or server (`force_client` and `force_hs`).
+
+##### Transparently onboarding new Matrix users
+
+New Matrix users following a matrix URI will typically expect to be onboarded
+to a specific branded client and/or server.  It is critical to minimise
+friction in doing so, in order to compete with the UX of traditional
+centralised apps. For instance, FOSDEM wants to onboard new users straight
+into https://chat.fosdem.org and default to using an account on the fosdem.org
+homeserver, via a single click.  To support this, we can specify query
+parameters to the URI to specify a default client and homeserver for new
+users:
+
+* `default_client` optionally specifies the client URI new users following this
+link should use to connect to Matrix, if there is not a registered matrix URI
+handler already.  To convert the matrix URL into the suggested default client
+URL, the caller should replace the matrix URI scheme prefix with the URL given
+by the `default_client` parameter, suffixed by `/#/` in order to express the
+routing data as a URL fragment and maximise privacy.  For instance, a
+matrix-URI aware browser with no registered matrix URI handler would convert
```

Comment:
yeah, I'll ask the Mozilla guys what their reaction would be to this.  However, i'm inclined to make it optional (which it already is with the 'should') and keep it in as a recommendation whatever, unless the feedback is "this will never happen".